prisma/schema.prisma

```diff
@@ -32,6 +32,7 @@ model Record {
   updatedAt   DateTime     @updatedAt
   expiresAt   DateTime
   status      RecordStatus
+  timesNotified Int    @default(0)
```

Comment:
Adding @cychu42 to this discussion for thoughts.

I'm not convinced about this approach (i.e., having a single integer count).  We have different types of notifications we want to send, and a count isn't going to cover them all.  I wonder if we should add a new table?

I don't have a solid proposal, but imagine something like this:

```prisma
model Notification {
  id            Int          @id @default(autoincrement())
  username      String
  sentAt        DateTime     @default(now())

  // This isn't perfect: we need to keep track of the record/cert this is associated with...
  certificateId Int?
  certificate   Certificate  @relation(fields: [certificateId], references: [id])

  recordId      Int?
  record        Record       @relation(fields: [recordId], references: [id])
}
```

Here we have the following info that we can use to track when we've sent notifications to users, specifically:

1. Who it is sent to (`user`)
2. When we sent it (`sentAt`)
3. What it relates to (a `certficateId` OR `recordId`).  There is probably a cleaner way to do this.  Do we need to use 2 tables instead?  Maybe that's better, but seems like a lot of extra infrastructure to me.

Using this, we could solve various problems. For example:

1. I need to send a notification about a specific expiring certificate to a user.  I can check to see if I've already sent one in the past X days, and either send it or ignore the request to send it based on the presence/absence of the notification record.
2. Similar to 1., I need to notify about an expiring record
3. Similar to 2., I need to notify about an expiring record that has already been renewed (e.g., there is already a notification record for this DNS record, but we need to send another one).  To do this, I can use the user + record + sent date, and figure out if I need to send another one (e.g., has it been X days or something).

I think we can improve on what I'm suggesting above, and hopefully @cychu42, @dadolhay and others can chime in with suggestions, too.

Reply:
Why don't we add a date field here, `notifyExpirationAt: Date`

When we create/renew a record, we set it 1 month before the expiry
If the date is in the past, we can send the notification, and mve this notifyExpirationAt to a future date or null if we don't want to send anymore emails

This would not require an extra table and related complexity

Reply:
@dadolhay, thanks for the feedback. I'd wait to hear @cychu42 thoughts on this and then implement the changes.

Reply:
I think the idea of using a date field is clever, since it might be enough to allow us to do this all in a the main tables (i.e. not adding new ones).

If it was a date of when we last notified a user about an expiry, we could do the following:

1. We add a `notifiedAt` field (name TBD) to the dns records and certs tables
2. When we create a record/cert, we'd leave it `notifiedAt` empty
3. When a record/cert is going to expire (e.g., 1 month out?) we send an email and update `notifiedAt` to current date time
4. When we renew a record/cert, we update the `expiry` and clear `notifiedAt`
5. The process repeats